cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -688,6 +688,8 @@ public void deleteArtifact(HttpRequest request, HttpResponder responder,
     try {
       artifactRepository.deleteArtifact(artifactId);
       responder.sendStatus(HttpResponseStatus.OK);
+    } catch (ArtifactNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Artifact " + artifactId + " not found.");
```

Comment:
I don't think you need to catch this. It should already be handled by the `HttpExceptionHandler` class.